agent/fw_drupal.c

```diff
@@ -588,6 +603,30 @@ NR_PHP_WRAPPER(nr_drupal_wrap_module_invoke) {
 }
 NR_PHP_WRAPPER_END
 
+#if ZEND_MODULE_API_NO >= ZEND_8_0_X_API_NO \
+     && !defined OVERWRITE_ZEND_EXECUTE_DATA
+NR_PHP_WRAPPER(nr_drupal_wrap_module_invoke_all_before) {
+  (void)wraprec;
+  zval* hook_copy = NULL;
+
+  NR_PHP_WRAPPER_REQUIRE_FRAMEWORK(NR_FW_DRUPAL);
+
+  hook_copy = nr_php_arg_get(1, NR_EXECUTE_ORIG_ARGS TSRMLS_CC);
+  if (nr_php_is_zval_non_empty_string(hook_copy)) {
+    nr_stack_push(&NRPRG(drupal_module_invoke_all_hooks),
+                  hook_copy);
+  }
+}
+NR_PHP_WRAPPER_END
+
+NR_PHP_WRAPPER(nr_drupal_wrap_module_invoke_all_after) {
+    (void)wraprec;
+    zval* hook_copy = nr_stack_pop(&NRPRG(drupal_module_invoke_all_hooks));
```

Comment:
What happens if on line 615 above it is decided to not push due to `hook_copy` being an empty string.  Will a value get popped in the "after" call that shouldn't?

Reply:
Yeah, good find. I'm trying to think of ways to solve this and can only think of 2 viable solutions. 1) keep a second stack of bools that indicates whether or not the current call has pushed to the drupal_module_invoke_all_hooks stack 2) push to the stack even when the string is null.empty and then when we are accessing the drupal_module_invoke_all_hooks, walk up the stack until we find a non-empty string.

I am currently leaning towards the first.

Reply:
I suppose the other option is 3) copy the hook from the args again in the after function to check if it's empty. This would create an unnecessary copy for the sake of code simplicity and readability.

I am actually pretty compelled to go with option 3 if nobody has issue with its tradeoffs.

Reply:
I actually opted for 3) but using the nr_php_get_user_func_arg to prevent the additional copy.

Walking up the stack would be necessary to maintain functionality with previous instrumentation. The module_invoke_all instrumentation was what maintains which value is set to the global variable (and refuses to add NULL/empty string) but this value is used in other instrumentations.

Reply:
wordpress check added here [1a451a7](https://github.com/newrelic/newrelic-php-agent/pull/559/commits/1a451a7af25aba4298dbc56c78f7341920d0a8b1)